include/deal.II/numerics/vector_tools.h

```diff
@@ -857,6 +858,71 @@ namespace VectorTools
                 const bool                          project_to_boundary_first = false);
 
   /**
+   * The same as above for projection of scalar-valued quadrature data.
+   * The user provided function should return a value at the quadrature point
+   * based on the cell iterator and quadrature number and of course should be
+   * consistent with the provided @p quadrature object, which will be used
+   * to assemble the right-hand-side.
+   *
+   * This function can be used with lambdas:
+   * @code
+   * VectorTools::project
+   * (mapping,
+   *  dof_handler,
+   *  constraints,
+   *  quadrature_formula,
+   *  [=] (const typename DoFHandler<dim>::active_cell_iterator & cell, const unsigned int q) -> double
+   *  { return qp_data.get_data(cell)[q]->density; },
+   *  field);
+   * @endcode
+   * where <code>qp_data</code> is a CellDataStorage object, which stores
+   * quadrature point data.
+   */
+  template <int dim, typename VectorType, int spacedim>
+  void project (const Mapping<dim, spacedim>   &mapping,
+                const DoFHandler<dim,spacedim> &dof,
+                const ConstraintMatrix         &constraints,
+                const Quadrature<dim>          &quadrature,
+                const std::function< typename VectorType::value_type (const typename DoFHandler<dim, spacedim>::active_cell_iterator &, const unsigned int)> func,
+                VectorType                     &vec_result);
```

Comment:
what is the header for `std_cxx11::function`? (i mean should use boost if compiler does not support c++11, as far as i recall).